### PR TITLE
Bug fix/7563 download solution from git cli

### PR DESCRIPTION
**Pull request by KashmiraDeshpande**

### CODE REVIEW

File: Ginger/Ginger/Repository/RepositoryItemFactory.cs [hunk 1]
```diff
@@ -744,8 +744,16 @@ public void DownloadSolution(string SolutionFolder)
             {
                 sol.ExistInLocaly = true;
             }
-            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT && Directory.Exists(PathHelper.GetLongPath(sol.LocalFolder + @"\.git")))
+            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT)
             {
+                if (!sol.LocalFolder.EndsWith(".git"))
+                {
+                    Directory.Exists(PathHelper.GetLongPath(sol.LocalFolder + @"\.git"));
```
Comment (JinendraGhodke) [inline Ginger/Ginger/Repository/RepositoryItemFactory.cs]: Directory.Exist was in the if condition earlier.
Now regardless if directory exist or not sol.existLocaly will be always marked true if Souce control type is git

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs [hunk 2]
```diff
@@ -37,23 +37,29 @@ public async void Execute(RunsetExecutor runsetExecutor)
...
         {
             string sConfig = null;
-            if (cliHelper.DownloadUpgradeSolutionFromSourceControl == true)
-            {               
-                sConfig = "SourceControlType=" + solution.SourceControl.GetSourceControlType.ToString() + Environment.NewLine;
-                sConfig += "SourceControlUrl=" + solution.SourceControl.SourceControlURL.ToString() + Environment.NewLine;
-                sConfig += "SourceControlUser=" + solution.SourceControl.SourceControlUser.ToString() + Environment.NewLine;
-                sConfig += "SourceControlPassword=" + solution.SourceControl.SourceControlPass.ToString() + Environment.NewLine;
-                if (solution.SourceControl.GetSourceControlType == SourceControlBase.eSourceControlType.GIT && solution.SourceControl.SourceControlProxyAddress.ToLower().ToString() == "true")
+            if (WorkSpace.Instance.Solution != null)
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs]: we are passing solution here explicitly. So not sure why relying on Workspace.
Either to remove the method param solution if not used

File: Ginger/Ginger/Repository/RepositoryItemFactory.cs [hunk 3]
```diff
@@ -744,8 +744,16 @@ public void DownloadSolution(string SolutionFolder)
             {
                 sol.ExistInLocaly = true;
             }
-            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT && Directory.Exists(PathHelper.GetLongPath(sol.LocalFolder + @"\.git")))
+            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT)
             {
+                if (!sol.LocalFolder.EndsWith(".git"))
+                {
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Repository/RepositoryItemFactory.cs]: Directory.Exists() lines are not part of the IF so i'm not getting how it is checking if git folder is exist or not and setting sol.ExistInLocaly flag accordinally

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs (hunk 2 above)
Comment (MeniKadosh1) [reply Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs]: do not remove the Solution param, the dependency on Workspace was removed in porpuse, why it was added back?

File: Ginger/GingerCoreNETUnitTest/RunTestslib/CLITest.cs [hunk 4]
```diff
@@ -368,7 +368,7 @@ public void TestRunsetAutoRunConfigCreationContent()
             executer.RunSetConfig = runsetConfig;
             CLIHelper cliHelp = new CLIHelper();
             RunSetAutoRunConfiguration autoRunConfig = new RunSetAutoRunConfiguration(sol, executer, cliHelp);
-
+            WorkSpace.Instance.Solution = sol;
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNETUnitTest/RunTestslib/CLITest.cs]: need instead to pass the Solution as param to RunSetAutoRunConfiguration constructor

File: Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs [hunk 5]
```diff
@@ -37,23 +37,29 @@ public async void Execute(RunsetExecutor runsetExecutor)
...
+            {
+                if (cliHelper.DownloadUpgradeSolutionFromSourceControl == true)
                 {
-                    sConfig += "SourceControlProxyServer=" + solution.SourceControl.SourceControlProxyAddress.ToString() + Environment.NewLine;
-                    sConfig += "SourceControlProxyPort=" + solution.SourceControl.SourceControlProxyPort.ToString() + Environment.NewLine;
+                    sConfig = "SourceControlType=" + solution.SourceControl.GetSourceControlType.ToString() + Environment.NewLine;
+                    sConfig += "SourceControlUrl=" + solution.SourceControl.SourceControlURL.ToString() + Environment.NewLine;
+                    sConfig += "SourceControlUser=" + solution.SourceControl.SourceControlUser.ToString() + Environment.NewLine;
+                    sConfig += "SourceControlPassword=" + solution.SourceControl.SourceControlPass.ToString() + Environment.NewLine;
+                    if (solution.SourceControl.SourceControlProxyAddress != null)
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs]: shift If's + add the validation for solution.SourceControl.SourceControlProxyAddress.ToLower().ToString() == "true" after checking it is not null

File: Ginger/Ginger/Repository/RepositoryItemFactory.cs [hunk 6]
```diff
@@ -744,9 +744,22 @@ public void DownloadSolution(string SolutionFolder)
             {
                 sol.ExistInLocaly = true;
             }
-            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT && Directory.Exists(PathHelper.GetLongPath(sol.LocalFolder + @"\.git")))
+            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT)
             {
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Repository/RepositoryItemFactory.cs]: 1- need to check which actual folder we try to pull the root folder of solution or the inner ".git" folder
2- test it for gitLab + gitHub solutions

**Pull request closed without merging**